### PR TITLE
[NEW] Limit all DDP/Websocket requests (configurable via admin panel)

### PR DESCRIPTION
**Pull request by rodrigok**

### CODE REVIEW

File: packages/rocketchat-lib/server/startup/rateLimite.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+RateLimiter.prototype.check = function(input) {
+	const self = this;
+	const reply = {
+		allowed: true,
+		timeToReset: 0,
+		numInvocationsLeft: Infinity,
+	};
+
+	const matchedRules = self._findAllMatchingRules(input);
+	_.each(matchedRules, function(rule) {
```
[the remark this answers is not in the document]
Comment (rodrigok) [reply packages/rocketchat-lib/server/startup/rateLimite.js]: I was keeping the meteor's code as much as possible

File: packages/rocketchat-lib/server/startup/rateLimite.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+const checkNameNonStream = (name) => name && !names.has(name) && !name.startsWith('stream-');
+const checkNameForStream = (name) => name && !names.has(name) && name.startsWith('stream-');
+
+const ruleIds = {};
+
+const callback = (message) => (reply, input) => {
+	if (reply.allowed === false) {
+		console.warn('DDP RATE LIMIT:', message);
```
[the remark this answers is not in the document]
Comment (rodrigok) [reply packages/rocketchat-lib/server/startup/rateLimite.js]: We don't have a level for warnings in Logger, do you think this message should be an error?

[the remark this answers is not in the document]
Comment (rodrigok) [reply packages/rocketchat-lib/server/startup/rateLimite.js]: @sampaiodiego but there is no way to hide errors as well 😄

[the remark this answers is not in the document]
Comment (sampaiodiego) [reply packages/rocketchat-lib/server/startup/rateLimite.js]: true :stuck_out_tongue:

**Pull request merged**